src/mono/mono/mini/interp/transform.c

```diff
@@ -6094,44 +6142,82 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			} else if (!td->optimized) {
 				int tos = get_tos_offset (td);
 				int param_offset, param_size;
-				if (csignature->param_count) {
-					td->sp -= csignature->param_count;
+				int param_count = csignature->param_count;
+				if (param_count) {
+					td->sp -= param_count;
 					param_offset = td->sp [0].offset;
 					param_size = tos - param_offset;
 				} else {
 					param_offset = tos;
 					param_size = 0;
 				}
 
-				td->cbb->contains_call_instruction = TRUE;
-				interp_add_ins (td, MINT_NEWOBJ_SLOW_UNOPT);
-				interp_ins_set_sreg (td->last_ins, MINT_CALL_ARGS_SREG);
-				init_last_ins_call (td);
-				td->last_ins->info.call_info->call_offset = param_offset;
-				td->last_ins->data [0] = get_data_item_index_imethod (td, mono_interp_get_imethod (m));
-				td->last_ins->data [1] = param_size;
-				if (csignature->param_count > 0) {
-					// Instruct opcode to also align params if necessary
-					if (td->locals [td->sp [0].local].flags & INTERP_LOCAL_FLAG_SIMD)
-						td->last_ins->data [3] = 1;
-				}
+				// Move params types in temporary buffer
+				StackInfo *sp_params = (StackInfo*) mono_mempool_alloc (td->mempool, sizeof (StackInfo) * param_count);
+				memcpy (sp_params, td->sp, sizeof (StackInfo) * param_count);
 
 				gboolean is_vt = m_class_is_valuetype (klass);
+				int vtsize = 0;
+				// We conservatively ensure stack size with an additional 2 * MINT_STACK_SLOT_SIZE since we can do
+				// at most 2 alignments
 				if (is_vt) {
-					int vtsize = mono_class_value_size (klass, NULL);
+					vtsize = mono_class_value_size (klass, NULL);
 					vtsize = ALIGN_TO (vtsize, MINT_STACK_SLOT_SIZE);
-					td->last_ins->data [2] = vtsize;
-					ENSURE_STACK_SIZE(td, (int)(tos + vtsize + MINT_STACK_SLOT_SIZE));
+					ENSURE_STACK_SIZE(td, (int)(tos + vtsize + MINT_STACK_SLOT_SIZE + 2 * MINT_STACK_SLOT_SIZE));
 					if (ret_mt == MINT_TYPE_VT)
 						push_type_vt (td, klass, vtsize);
 					else
 						push_type (td, stack_type [ret_mt], klass);
 				} else {
-					td->last_ins->data [2] = 0;
-					ENSURE_STACK_SIZE(td, (int)(tos + 2 * MINT_STACK_SLOT_SIZE));
+					ENSURE_STACK_SIZE(td, (int)(tos + 2 * MINT_STACK_SLOT_SIZE + 2 * MINT_STACK_SLOT_SIZE));
```

Comment:
maybe extract the '2 * MINT_STACK_SLOT_SIZE' into a named constant like 'padding' so it's clear what the distinction is between the pair of them here for later code maintenance?